### PR TITLE
Polishing the API and the docu. Dot notation and a couple getters.

### PR DESCRIPTION
**Pull request by dkorolev**

### CODE REVIEW

**Comment by dkorolev** [comment]
@sompylasar @deathbaba PTAL.

File: README.md [hunk 1]
```diff
@@ -242,39 +188,96 @@ struct PennyInput {
   std::vector<int> x;
   template <typename A> void serialize(A& ar) {
     ar(CEREAL_NVP(op), CEREAL_NVP(x));
-  }   
-  // TODO(dkorolev): Safe mode wrt parsing malformed JSON.
+  }
```
Comment (sompylasar) [inline README.md]: GitHub shows something red here and below (a space?).

File: README.md [hunk 2]
```diff
@@ -242,39 +188,96 @@ struct PennyInput {
   std::vector<int> x;
   template <typename A> void serialize(A& ar) {
     ar(CEREAL_NVP(op), CEREAL_NVP(x));
-  }   
-  // TODO(dkorolev): Safe mode wrt parsing malformed JSON.
+  } 
+  void FromInvalidJSON(const std::string& input_json) {
+    op = "JSON parse error: " + input_json;
```
Comment (sompylasar) [inline README.md]: Is this OK to put an error message into the "operation" field?

**Comment by dkorolev** [comment]
Fixed the whitespaces. Some are necessary for ```README.md``` indent, but certainly not all.

And you're right about the error message, let me fix that one too in a sec.

**Comment by dkorolev** [comment]
Fixed, PTAL.

File: README.md [hunk 3]
```diff
@@ -240,41 +183,100 @@ HTTP(port).Register("/found", [](Request r) {
...
+      for (const auto v : input.x) {
+        result += v;
+      }
+    } else if (input.op == "mul") {
+      result = 1;
+      for (const auto v : input.x) {
+        result *= v;
+      }
     }
+    r(PennyOutput{"", result});
```
Comment (sompylasar) [inline README.md]: Was there some sort of unknown operation error few minutes ago?

Comment (dkorolev) [inline README.md]: There was no, but I just added it while you've been looking at this code :)

Comment (dkorolev) [inline README.md]: Just added a test. On the other hand, we also need a "not enough parameters" case, give me a sec.

File: net/http/impl/server.h [hunk 4]
```diff
@@ -505,9 +523,10 @@ class HTTPServerConnection final {
     std::unique_ptr<Impl> impl_;
   };
 
-  inline ChunkedResponseSender SendChunkedHTTPResponse(HTTPResponseCode code = HTTPResponseCode::OK,
-                                                       const std::string& content_type = DefaultContentType(),
-                                                       const HTTPHeaders& extra_headers = HTTPHeaders()) {
+  inline ChunkedResponseSender SendChunkedHTTPResponse(
+      HTTPResponseCodeValue code = HTTPResponseCodeValue::OK,
```
Comment (sompylasar) [inline net/http/impl/server.h]: `HTTPResponseCodeValue::OK` -> `HTTPResponseCode.OK` (like everywhere else)

**Comment by dkorolev** [comment]
Just pushed Penny's logic to be real, with the tests. Thanks for pointing it out, PTAL.

**Comment by dkorolev** [comment]
@sompylasar : Good eye! That was the very only last one left :-) Fixed it too.

**Comment by dkorolev** [comment]
(Also, you should really go to bed ...)

**Comment by sompylasar** [comment]
Yep... The birds are singing...

Right, this one is good to merge if the tests are passing.

**Comment by dkorolev** [comment]
Of course, they are, with 100% coverage :) Thanks for a late night / early morning review, really appreciate it.

**Pull request merged**